app/controllers/status_messages_controller.rb

```diff
@@ -1,128 +1,97 @@
 #   Copyright (c) 2010-2011, Diaspora Inc.  This file is
 #   licensed under the Affero General Public License version 3 or later.  See
 #   the COPYRIGHT file.
-
 class StatusMessagesController < ApplicationController
   before_action :authenticate_user!
 
-  before_action :remove_getting_started, :only => [:create]
+  before_action :remove_getting_started, only: :create
 
-  respond_to :html,
-             :mobile,
-             :json
+  respond_to :html, :mobile, :json
 
-  layout 'application', only: :bookmarklet
+  layout "application", only: :bookmarklet
 
   # Called when a user clicks "Mention" on a profile page
   # @param person_id [Integer] The id of the person to be mentioned
   def new
-    if params[:person_id] && @person = Person.where(:id => params[:person_id]).first
+    if params[:person_id] && get_person(params[:person_id])
       @aspect = :profile
       @contact = current_user.contact_for(@person)
-      @aspects_with_person = []
       if @contact
         @aspects_with_person = @contact.aspects
-        @aspect_ids = @aspects_with_person.map{|x| x.id}
+        @aspect_ids = @aspects_with_person.map(&:id)
         gon.aspect_ids = @aspect_ids
-        @contacts_of_contact = @contact.contacts
-        render :layout => nil
+        render layout: nil
+      else
+        @aspects_with_person = []
       end
-    elsif(request.format == :mobile)
+    elsif request.format == :mobile
       @aspect = :all
       @aspects = current_user.aspects
-      @aspect_ids = @aspects.map{ |a| a.id }
+      @aspect_ids = @aspects.map(&:id)
       gon.aspect_ids = @aspect_ids
     else
       redirect_to stream_path
     end
   end
 
+  def get_person(person_id)
+    @person = Person.where(id: person_id).first
+  end
+
   def bookmarklet
     @aspects = current_user.aspects
-    @aspect_ids = @aspects.map{|x| x.id}
+    @aspect_ids = @aspects.map(&:id)
```

Comment:
And thinking a bit more about it, Rails even has a convenience method for the plucking on the association, `current_user.aspect_ids`.